includes/class-salesforce.php

```diff
@@ -28,6 +29,544 @@ class Salesforce {
+		register_rest_route(
+			self::SALESFORCE_API_NAMESPACE,
+			'/sync',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_sync_salesforce' ],
+				'permission_callback' => [ __CLASS__, 'api_validate_webhook' ],
+			]
+		);
+
+		// Legacy namespace handler.
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/salesforce/sync',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_sync_salesforce' ],
+				'permission_callback' => [ __CLASS__, 'api_validate_webhook' ],
+			]
+		);
+	}
+
+	/**
+	 * Check capabilities for using API.
+	 *
+	 * @return bool|WP_Error
+	 */
+	public static function api_permissions_check() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return new \WP_Error(
+				'newspack_rest_forbidden',
+				esc_html__( 'You cannot use this resource.', 'newspack' ),
+				[
+					'status' => 403,
+				]
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Check validity of webhook requester before processing the webhook callback handler.
+	 *
+	 * @param WP_REST_Request $request API request object.
+	 *
+	 * @return bool}WP_Error
+	 */
+	public static function api_validate_webhook( $request ) {
+		$is_valid   = false;
+		$webhook_id = $request->get_header( 'X-WC-Webhook-ID' );
+
+		if ( $webhook_id ) {
+			$is_valid = (int) get_option( self::SALESFORCE_WEBHOOK_ID, 0 ) === (int) $webhook_id;
+		}
+
+		if ( $is_valid ) {
+			return true;
+		}
+
+		return new \WP_Error(
+			'newspack_rest_forbidden',
+			esc_html__( 'Invalid webhook request.', 'newspack' ),
+			[
+				'status' => 403,
+			]
+		);
+	}
+
+	/**
+	 * Get settings list.
+	 *
+	 * @return WP_REST_Response containing the settings list.
+	 */
+	public static function api_get_settings() {
+		return \rest_ensure_response( self::get_settings_list( true ) );
+	}
+
+	/**
+	 * Get settings list.
+	 *
+	 * @param boolean $assoc If true, return settings as an associative array keyed by setting key, otherwise return as a flat array.
+	 *
+	 * return array Array of settings.
+	 */
+	private static function get_settings_list( $assoc = false ) {
+		$settings      = self::get_salesforce_settings();
+		$client_key    = $settings['client_id'];
+		$client_secret = $settings['client_secret'];
+		$settings_list = [
+			[
+				'description' => __( 'Connected App Settings', 'newspack' ),
+				'help'        => __( 'Settings for your Salesforce connected app.', 'newspack' ),
+				'section'     => 'salesforce',
+				'key'         => 'active',
+				'type'        => 'boolean',
+				'default'     => true,
+				'public'      => true,
+				'value'       => null,
+			],
+			[
+				'description' => __( 'Consumer Key', 'newspack' ),
+				'help'        => __( 'The connected app’s Consumer Key.', 'newspack' ),
+				'section'     => 'salesforce',
+				'key'         => self::SALESFORCE_CLIENT_ID,
+				'type'        => 'string',
+				'default'     => true,
+				'public'      => true,
```

Comment:
The `public` attribute determines whether that setting value is safe to be available to the public.

Reply:
Ah, thanks for clarifying! Changed this to `false` for all settings in bcde8d41.